consensus/carnot-engine/src/overlay/tree_overlay/overlay.rs

```diff
@@ -146,10 +146,14 @@ where
         if self.is_member_of_leaf_committee(id) {
             return 0;
         }
-        self.carnot_tree
+        match self
+            .carnot_tree
             .committee_by_member_id(&id)
             .map(|c| apply_threshold(c.len(), self.threshold))
-            .expect("node is not part of any committee")
+        {
+            Some(threshold) => threshold,
+            None => panic!("node {id} is not part of any committee"),
+        }
```

Comment:
Why is there a change in the carnot tree in this PR?

Reply:
Ah, this is used when I debugging, forget to change it back.

Reply:
This is still here. As well as other few changes and nitpicks.